core/arch/arm/kernel/boot.c

```diff
@@ -1196,7 +1196,11 @@ static void discover_nsec_memory(void)
 	const struct core_mmu_phys_mem *mem_begin = NULL;
 	const struct core_mmu_phys_mem *mem_end = NULL;
 	size_t nelems;
+#ifdef CFG_EXTERNAL_DT
 	void *fdt = get_external_dt();
+#else /* CFG_EXTERNAL_DT */
+	void *fdt = get_embedded_dt();
+#endif /* CFG_EXTERNAL_DT */
```

Comment:
prefer:
```patch
 	void *fdt = get_external_dt();
 
+	if (!fdt)
+		fdt = get_embedded_dt();
 
 	if (fdt) {
```

I have some trouble with this sequence: here we try external and fallback to embedded. That is not very consistent with [`configure_console_from_dt()`](https://elixir.bootlin.com/op-tee/3.18.0/source/core/kernel/console.c#L129) and other places calling [`get_dt()`](https://elixir.bootlin.com/op-tee/3.18.0/source/core/arch/arm/kernel/boot.c#L592) that does the opposite.
As we have quite a legacy here on getting DRAM info from external DT, I would not replace line 1200 with a call to `get_dt()`. Since platform can use `register_ddr()` to have these information embedded, I think this change is not mandated by introduction of the new config switch `CFG_EXTERNAL_DT` and should be further discussed.

Reply:
Ok, dropping this patch

Reply:
IMO, it is not an issue if it falls back every time when CFG_EXTERNAL_DT=n since every platform have these configs embedded.

Maybe it should be explicitly said to be careful to double-check the embedded values as there will be no further fallback